### ISSUE TITLE
Extend dials.symmetry to include all 230 crystallographic space groups

### ISSUE THREAD

**Issue by benjaminhwilliams** [dials/dials#1118]
To date, `dials.symmetry` only knows about the 65 chiral space groups.  This limits its utility somewhat in treatment of small molecule data.  We should teach it about the other 165.

@jbeilstenedmands' initial work on this suggests that the method previously used for the chiral groups, of tabulating a hard-coded dictionary of symmetry operations by group, is too laborious and prone to human error.  I'm sure this is not a novel task.  We should find out how it's been done before and emulate the best/lowest-effort options.

**Comment by dagewa** [comment]
I completely agree we should not be limited to 65. What exactly is required here? Getting hold of the symmetry operations of any space group is not  difficult. I was working on this today to produce SHELX records:
```
from cctbx import sgtbx
from iotbx.shelx.write_ins import LATT_SYMM
import sys
space_group_info = sgtbx.space_group_info(symbol="I a -3 d")
LATT_SYMM(sys.stdout, space_group_info.group())
```

**Comment by dagewa** [comment]
There are some notes here on various approaches taken by other packages: https://gemmi.readthedocs.io/en/latest/sym.html#implementation-notes

**Comment by jbeilstenedmands** [comment]
Thanks for the link David, I think the Gemmi implementation notes summarise the issue well:
"The Hall notation (-I 4bd 2c 3), first proposed by Sydney R. Hall in 1981, is shorter and more popular. It can be interpreted by a few libraries:..... On the bad side, the conciseness is achieved by complex rules of interpreting the symbols; the choice of a Hall symbol for given settings is not unambiguous and the symbols differ between editions of ITfC, and between sgtbx and syminfo.lib." 

I was having a look in cctbx and found this example of symmetry stuff that is very relevant to this problem: `cctbx\examples\find_sys_abs_equiv_space_groups.py`. However, I can not follow what is going on in the code. I'm sure we can use cctbx to do what we want.

For our problem, I guess what we want is;
- For the Laue group, generate all possible space groups
- For each space group, generate the systematic absence rules due to each relevant element (glides and screws), taking care of any absences due to lattice centering.
- Score these (straightforward) and combine the scores to score the space group
- May also need to handle non-standard settings. Most obvious example is Pmmm, as axes can be in any order, so 6 permutations for e.g. SG33 Pna21. Most obvious thing is to reindex to standard setting and then reindex back if we want to keep the input order of axes.

A slightly alternative approach, which is the approach I had taken for the MX groups:
- For each Laue group, generate all possible screw/glide operations (I manually listed these for MX, difficult to do for all space groups).
- Score these (straightforward)
- Combine operations/scores to generate space groups and scores (straightforward for MX, complicated for all space groups).

**Comment by biochem-fan** [comment]
Just wanted to say I would also like to have this feature for small molecule MicroED.

**Comment by graeme-winter** [comment]
@benjaminhwilliams  @jbeilstenedmands what do we think the correct behaviour should be for e.g. `P22_12`? What do we do at the moment?

I would think - 
 - reindex according to the Laue / Patterson symmetry e.g. rotational operators only
 - assign the space group in a potentially non-standard setting

This is then robust against presence or absence of a couple of axial reflections. Obviously for something orthorhombic with axial absences the list of possible space groups will be long when adding the nonchiral space groups.

**Comment by jbeilstenedmands** [comment]
I think what you suspect is correct. We leave the cell in its input setting (doing the Patterson symmetry in the 'standard' cell on a copy of the data), with the axial reflections tested in the input setting. We decided this is what we wanted when writing the program, leaving it up to the user to manually reindex into the standard setting if desired.

**Comment by biochem-fan** [comment]
I also agree with your suggestion.

**Comment by benjaminhwilliams** [comment]
That sounds reasonable.

**Comment by dagewa** [comment]
Just a note to say I am looking into this in the context of the I19 project. I am trying to understand the current approach in `dials.symmetry` and `dials.cosym` (and finding it difficult to follow). The `sg` command of Olex2 is worth a look for a cctbx-powered approach.

**Comment by jbeilstenedmands** [comment]
Feature now added and in the dials 3.29 release
